gtsfm/runner/gtsfm_runner_base.py

```diff
@@ -86,11 +106,30 @@ def construct_scene_optimizer(self) -> SceneOptimizer:
 
         return scene_optimizer
 
+    def construct_retriever(self):
+        """Set up retriever module."""
+        if self.parsed_args.matching_regime == "exhaustive":
+            retriever = SequentialRetriever(max_frame_lookahead=MAX_POSSIBLE_FRAME_LOOKAHEAD)
+
```

Comment:
I think it would be cleaner if this was it's own class.

Reply:
Gotcha -- could you share a bit more about the overall structure you're envisioning?

Reply:
It could just be a child of the `SequentialRetriever` and it would set `max_frame_lookahead=MAX_POSSIBLE_FRAME_LOOKAHEAD` by default.

Reply:
Since exhaustive matching can occur outside of sequential image sets, I see it as separate from the SequentialRetriever (except that SequentialRetriever allows for reuse of code).